packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -333,12 +336,62 @@ abstract class IntelliJValidator extends DoctorValidator {
 
       return;
     }
+    messages.add(new ValidationMessage.error(
+      '$title plugin not installed; this adds $title specific functionality.'
+    ));
+  }
+
+  Future<Null> _validateFlutterPackage(List<ValidationMessage> messages, List<String> packageNames, String title, {
+    Version minVersion
+  }) async {
 
+    for (String packageName in packageNames) {
+      if (!hasPackage(packageName)) {
+        continue;
+      }
+      final String versionText = _readPackageVersion(packageName);
+      final Version localVersion = new Version.parse(versionText);
+      if (localVersion != null && minVersion != null && localVersion < minVersion) {
+        messages.add(new ValidationMessage.error(
+          '$title plugin version $versionText - the recommended minimum version is $minVersion'
+        ));
+      } 
+      else {
+        
+        try {
+          final String latestFlutter = await _getLatestVersionPossible();
+          final Version flutterVersion = new Version.parse(latestFlutter);
+          final String serverMessage = localVersion < flutterVersion ? '[Consider upgrading your plugin version to $latestFlutter]' : '';
+            messages.add(new ValidationMessage(
+              '$title plugin ${localVersion != null ? "version $localVersion" : "installed"} $serverMessage'
+            ));
+        } catch (e) {          
+          messages.add(new ValidationMessage(
+            '$title plugin ${localVersion != null ? "version $localVersion" : "installed"} [$e]'
+          ));
+        }
+      }
+
+      return;
+    }
     messages.add(new ValidationMessage.error(
       '$title plugin not installed; this adds $title specific functionality.'
     ));
   }
 
+  Future<String> _getLatestVersionPossible() async {
+      final String url = 'https://plugins.jetbrains.com/plugins/list?build=$buildNumber&pluginId=io.flutter';
+
+      try {
+        final http.Response response = await http.get(url);
+        final xml.XmlDocument xmlDocument = xml.parse(response.body);
+        final String version = xmlDocument.findAllElements('version').map((xml.XmlElement node) => node.text).first;
+        return version;
+      } catch (e) {
+        throw 'Unable to check for the plugin updates in the plugin repository';
```

Comment:
[Nit] maybe shorten to:

    'Unable to check plugin repository for updates'

?

Reply:
Also, maybe we should add the exception message too?

Reply:
Ok going to use that short version. 
About the exception If i print the exception message if would probably make the console output a bit ugly when offline for instance:
![image](https://user-images.githubusercontent.com/5158380/34422407-4a6a9dd6-ebe3-11e7-8e76-8be814724425.png)

If that is ok i can include it. Unless we print the exception message only on verbose mode so you can get more details. Let me see how to get the global flag.